README.md

```diff
@@ -15,4 +15,22 @@ results you would get.
 Also give a theoretical argument for why X could or could not be correct, based
 on the complexity of the general sorting problem we covered in class.
 
-Add your answers to this markdown file.
+
+# My Answer
+
+Hearing of such an algorithm would make me quite skeptical at first. Skeptical enough for me to believe that this algorithm doesn't actually exist.
+
+To verify my claim, I would put the algorithm through some rigorous testing to see if I can either make it break or show any runtimes that would suggest its runtime is greater than O(n). I would run it through various unit tests with all sorts of data types and different combinations of data. And I would come up with as many property tests as I could to verify it all as well. Comparing it against other fast algorithms and graphing the results would also be something to look at - trying to deduce if anything looks suspicious.
+
+Those are all the concrete ways to try and see if the algorithm truly is linear, however, let's consider some other things conceptually. In order for any sorting algorithm to make decisions, it needs to go through and check *at least* every element in the array - otherwise we wouldn't be able to confirm whether the list is sorted or not. We need to be able to "see" the data. 
+
+Now consider this linear-runtime algorithm. As the algorithm goes along and checks some elements, there are two different parts of the array. The part we have "seen" and the part we have "not seen." If this algorithm is linear and is sorting elements as it goes, it is placing elements blind. The odds of misplacing elements are (almost) gaurunteed because there is no way to know what is coming up next - the algorithm is essentially forced to guess where to put the elements. (There is a very small possibility where it could correctly sort an array, but this would be an incredibly rare event depending on the size of the array - especially if the input is large). Because of this, I would highly doubt that the algorithm is actually linear.
```

Comment:
Not following you here. Going through all the elements is linear, so it would definitely "see" all the elements.

Reply:
This is assuming the algorithm is in the middle of execution. 

When the algorithm begins, we have not yet processed any elements (the "not seen" portion of the array). As we go along and start to compare elements, you could think of those elements as "seen". At the end where we finish up comparing all the elements, we will have "seen" the entire array. 

My point here is saying that we don't know where elements should go if we have not yet observed them.

Reply:
But there is (asymptotically) enough time to observe them. Think about the theoretical sorting lecture, and the argument we made to establish a lower bound for all comparison-based sorting algorithms.